### PR TITLE
test: add e2e test cases for upstream

### PR DESCRIPTION
**Pull request by nic-chen**
add e2e test cases for upstream

and fix bugs found during testing

### CODE REVIEW

**Review by membphis — CHANGES_REQUESTED** [review]
as we talked before, one PR for one thing, if you found new bug, we need to fix them in a new PR.

**Comment by membphis** [comment]
we should only add E2E code in this PR

File: api/internal/handler/upstream/upstream.go [hunk 1]
```diff
@@ -144,8 +146,33 @@ type BatchDelete struct {
 
 func (h *Handler) BatchDelete(c droplet.Context) (interface{}, error) {
 	input := c.Input().(*BatchDelete)
+	ids := strings.Split(input.IDs, ",")
 
-	if err := h.upstreamStore.BatchDelete(c.Context(), strings.Split(input.IDs, ",")); err != nil {
+	for _, id := range ids {
```
Comment (membphis) [inline api/internal/handler/upstream/upstream.go]: why do we need those codes?

they are not E2E test codes.

Comment (nic-chen) [reply api/internal/handler/upstream/upstream.go]: removed.

**Review by membphis — APPROVED** [review]
LGTM, good luck

**Comment by membphis** [comment]
merged, many thx

**Pull request merged**